mlflow/models/evaluation/base.py

```diff
@@ -264,6 +264,7 @@ def __init__(self, data, *, targets, name=None, path=None, feature_names=None):
                 spark_df_type = None
         except ImportError:
             supported_dataframe_types = (pd.DataFrame,)
+            spark_df_type = None
```

Comment:
Can we simplify the code here?


```python
        spark_df_type = None
        supported_dataframe_types = (pd.DataFrame,)
        try:
            # add checking `'pyspark' in sys.modules` to avoid importing pyspark when user
            # run code not related to pyspark.
            if "pyspark" in sys.modules:
                from pyspark.sql import DataFrame as SparkDataFrame
                supported_dataframe_types = (*supported_dataframe_types, SparkDataFrame)
                spark_df_type = SparkDataFrame
        except ImportError:
            pass
```

Reply:
Great point! Pushed a change so that the reassignment only happens in one place.